internal/pipeline/pipeline.go

```diff
@@ -0,0 +1,55 @@
+package pipeline
+
+import (
+	"fmt"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/domain/status"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/usecases/link"
+	"net/http"
+	"sync"
+	"time"
+)
+
+func LinkStatusUpdater(luc *link.LinkUseCases) {
+	go func() {
+		links := make(chan link.Link)
+
+		go func() {
+			for {
+				userLinks, err := luc.LinkStorage.GetAllUserLinks()
+				if err != nil {
+					panic(err)
+				}
+				for _, lnk := range userLinks {
+					links <- link.Link{LinkId: lnk.LinkId, Link: lnk.Link, LinkStatus: lnk.LinkStatus}
+				}
+				time.Sleep(5 * time.Second)
+			}
+		}()
+
+		var wg sync.WaitGroup
+		for i := 0; i < 4; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for {
+					lnk, ok := <-links
+					if !ok {
+						return
+					}
+					res, _ := http.Get(lnk.Link)
+					s := status.OK
+					if res.StatusCode == http.StatusNotFound {
+						s = status.Failed
+					}
+					_ = luc.LinkStorage.UpdateLinkStatusByLinkId(lnk.LinkId, s)
```

Comment:
You are just ignoring the possible problem and messaging about success?

Reply:
Now we are not ignoring an error.

Reply:
Ok, I accept it. Nevertheless, it seems a little bit strange to write "%v status changed from %v to %v\n" right after an error.

Reply:
Fixed.